lfs/gitscanner_catfilebatch.go

```diff
@@ -41,8 +42,10 @@ func runCatFileBatch(pointerCh chan *WrappedPointer, lockableCh chan string, loc
 			}
 		}
 
-		if err := revs.Wait(); err != nil {
-			errCh <- err
+		if canScan {
```

Comment:
I'm unfamiliar with this code so I'm a bit confused as to how this accomplishes what's listed in the commit message. Could you elaborate?

Reply:
Sure. If, for some reason, the scanner fails, such as running out of file descriptors, it returns false from `Scan`. In such a case, the thread producing values on `revs.Results` will be expecting us to read more objects from the data channel and will not have closed the error channel. However, this thread isn't reading anything from the data channel, and is waiting on the error channel, and will continue to do so until it's closed.

Since this thread is waiting on the error channel to have data or be closed, and the other thread is waiting to write to the data channel, we deadlock whenever `Scan` returns false. This prevents the return of the useful error message to the user and results in a hang; therefore, this commit is more about returning a useful error message in case of failure instead of hanging.

Reply:
Ah okay, that makes a lot of sense. Thanks for the info!